pkg/controller/direct/containerattached/attachedcluster_controller.go

```diff
@@ -0,0 +1,342 @@
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/registry"
+
+	gcp "cloud.google.com/go/gkemulticloud/apiv1"
+
+	containerattachedpb "cloud.google.com/go/gkemulticloud/apiv1/gkemulticloudpb"
+	"github.com/googleapis/gax-go/v2/apierror"
+	"google.golang.org/api/option"
+	// "google.golang.org/grpc/credentials"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/status"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/known/fieldmaskpb"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func init() {
+	registry.RegisterModel(krm.ContainerAttachedClusterGVK, NewContainerAttachedClusterModel)
+}
+
+func NewContainerAttachedClusterModel(ctx context.Context, config *config.ControllerConfig) (directbase.Model, error) {
+	return &modelContainerAttachedCluster{config: *config}, nil
+}
+
+var _ directbase.Model = &modelContainerAttachedCluster{}
+
+type modelContainerAttachedCluster struct {
+	config config.ControllerConfig
+}
+
+func loggingUnaryInterceptor() grpc.UnaryClientInterceptor {
+	return func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+			err := invoker(ctx, method, req, reply, cc, opts...)
+			log.Printf(":HF: Invoked method: %v, %+v", method, err)
+			md, ok := metadata.FromOutgoingContext(ctx)
+			log.Printf(":HF: md: %+v, ok: %t", md, ok)
+			if ok {
+					log.Println("Metadata:")
+					for k, v := range md {
+							log.Printf("Key: %v, Value: %v", k, v)
+					}
+			}
+			reqb, merr := protojson.Marshal(req.(protoreflect.ProtoMessage))
+			if merr == nil {
+					log.Printf(":HF: Request: %s", reqb)
+			}
+			log.Printf(":HF: merr: %+v", merr)
+			return err
+	}
+}
+
+func (m *modelContainerAttachedCluster) client(ctx context.Context, endpoint string) (*gcp.AttachedClustersClient, error) {
+	var opts []option.ClientOption
+	// Not working ("WithHTTPClient is incompatible with gRPC dial options"). GRPCClientOptions() gives the same error, and
+	// the implementation looks functionally identical.
+	// opts, err := m.config.GRPCClientOptions()
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	// Added an interceptor to add more logging, but couldn't find anything useful.
+	// opts = append(opts, option.WithGRPCDialOption(grpc.WithUnaryInterceptor(loggingUnaryInterceptor())))
+	opts = append(opts, option.WithEndpoint(endpoint))
+	gcpClient, err := gcp.NewAttachedClustersClient(ctx, opts...)
+	// gcpClient, err := gcp.NewAttachedClustersClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("building AttachedCluster client: %w", err)
+	}
+	return gcpClient, err
+}
+
+func endpoint(location string) string {
+	return fmt.Sprintf("%s-gkemulticloud.googleapis.com:443", location)
+}
```

Comment:
hey @justinsb / @yuwenma any good leads on why we are not intercepting the client's http traffic here?

AFAICT, the mock is not being called using this direct ctrl and it's possibly bc of how we build the gcpClient